packages/primitives/link/src/Root.native.tsx

```diff
@@ -21,10 +21,20 @@ export const Link = component(
       }
     },
   })
-)(({ children, id, onPress }) => (
-  <Text testID={id} onPress={onPress}>
-    {children}
-  </Text>
-))
+)(({ children, id, onPress }) => {
+  if (typeof children === 'string' || typeof children === 'number') {
+    return (
+      <Text testID={id} onPress={onPress}>
+        {children}
+      </Text>
+    )
+  }
+
+  return (
+    <TouchableWithoutFeedback testID={id} onPress={onPress}>
```

Comment:
Should `TouchableWithoutFeedback` have `accessibilityRole="link"` here?

Reply:
it seems like in bubble-ui we use `accessibilityLabel` but YES, it seems that this component can be use some accessibility, both here and Text and web

Reply:
`accessibilityRole` and `accessibilityLabel` are two different things (as I learned today :D).
It seems we should set the `accessibilityRole="link"` so the screen readers read "link" instead of "touchable without feedback" when they encounter this component AND we should allow to pass the `accessibilityLabel` to the `TouchableWithoutFeedback` from the outside in case the link is just an icon.
What do you think?

Reply:
https://reactnative.dev/docs/accessibility#accessibilitylabel-android-ios

https://reactnative.dev/docs/accessibility#accessibilityrole-android-ios

yes! accessibilityRole is the way to go!
added it just now

Reply:
not sure about adding the `accessibilityLabel` i think who ever uses the primitive should handle the accessibility, cause it seems the most inner component accessibility will take over

Reply:
> i think who ever uses the primitive should handle the accessibility
Yes, but we need to allow them to do it, since we're not spreading the props :)
So, I thought about adding `accessibilityLabel` as optional prop that can be passed from outside to `TouchableWithoutFeedback`. So we don't hardcode anything, just allow to pass it in case someone decides that have a link with only an icon and needs to add some explanation for screen readers.
I also agree that the `accessibilityLabel`s should be passed down to primitive level like `id`s.
So, I think we agree on that? Or did I misunderstand your answer?

Reply:
i guess no harm in that one, added!